pkg/detectors/ldap/ldap.go

```diff
@@ -98,7 +108,19 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 	return results, nil
 }
 
-func verifyLDAP(ctx context.Context, username, password string, ldapURL *url.URL) bool {
+func isErrDeterminate(err error) bool {
+	switch e := err.(type) {
+	case *ldap.Error:
+		switch e.Err.(type) {
+		case *net.OpError:
+			return false
+		}
+	}
+
+	return true
+}
```

Comment:
I tried changing this function to the following:

```go
func isErrDeterminate(err error) bool {
	var target *net.OpError
	return !errors.As(err, &target)
}
```

and it passed the tests, though this function also passed the tests:

```go
func isErrDeterminate(err error) bool {
	return true
}
```

`errors.As` is a bit more flexible since it will `Unwrap` errors.

Reply:
err, that second function shouldn't have passed...

Reply:
sorry for the confusion, I didn't run the integration tests 😕